app/controllers/recipes_controller.rb

```diff
@@ -1,24 +1,18 @@
 class RecipesController < ApplicationController
   def index
-    recipes = SpoonacularService.new(search_params).parse_recipes
-    @recipes = recipes[:results].map do |recipe_json|
-      Recipe.new(recipe_json)
-    end
-    if @recipes.empty?
-      flash[:error] = "Sorry, we couldn't find any recipes matching your
-      specification. Here's some random recipes you might like."
-      redirect_to '/recipes'
-    end
+    @recipes = Recipe.search(search_params)
+    return unless @recipes.empty?
+
+    flash[:error] = "Sorry, we couldn't find any recipes matching your
+    specification. Here are some random recipes you might like."
+    redirect_to recipes_path
   end
 
   def show; end
 
   private
 
   def search_params
-    params.permit(:ingredients,
-                  :time,
-                  :type,
-                  diet: [])
+    params.permit(:ingredients, :time, :type, :diet)
```

Comment:
Was the array unnecessary? I don't think there's a specific test in place if for example you pass in dairy free, gluten free and vegan, but I was running into the issue where the parameters would overwrite the :diet key.. The array actually saves each of the checked boxes into the array

Reply:
The diet values still do post to SearchController#create as an array in the params. Since the SpoonacularService had been joining the array into a string for passing in as an API query parameter, I figured we might as well convert it as soon as it's posted (see SearchController line 10).

In checking this out, though, I did notice that I introduced a bug where not checking any prep time option resulted in the API query parameter specifying a max prep time of zero minutes lol. Fixed that!

Reply:
Not sure if this will interfere with the functionality I built in for automatically checking boxes.... hopefully it doesn't but we'll find out.  I'm hoping since things are being targeted with xpath (Thanks David!) they will be found even without diet[]